jenkins/python-cache-cleanup.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+# We delete any python compiler caches to avoid conflicts with the python compiler
+# in the container.
+find . -name __pycache__ -exec rm -r {} \; > /dev/null 2>&1
```

Comment:
> Consider replacing the `\;` with `\+`.

What do you think the `\;` is doing that we could consider using `\+`?  See the definition of the `-exec` switch at https://man7.org/linux/man-pages/man1/find.1.html.  The exec'd command needs to end with a semi-colon passed to `find` and not interpreted by `bash`.  So the `\;` helps the semi-colon escape `bash`'s clutches.

Reply:
With `\;`, the `{}` is replaced by a single found-item, while with `\+` it is replaced by a list of found items, thereby requiring fewer invocations of the `-exec`'d command.  See the (second) definition of the -exec switch at https://man7.org/linux/man-pages/man1/find.1.html.  😉

Reply:
> With `\;`, the `{}` is replaced by a single found-item, while with `\+` it is replaced by a list of found items, thereby requiring fewer invocations of the `-exec`'d command. See the (second) definition of the -exec switch at https://man7.org/linux/man-pages/man1/find.1.html. wink

I'd never encountered the `\+` syntax before, which is really cool. I quite frequently pipe `find` into `xargs` rather than using the built-in `-exec` for exactly that reason.